epubmaker_postprocessing.rb

```diff
@@ -43,119 +120,180 @@ def listSpacebreakImages(file)
   path_to_i = File.join(dir, file)
   myres = `identify -format "%y" "#{path_to_i}"`
   myres = myres.to_f
   if File.file?(path_to_i)
     puts "RESIZING #{path_to_i} for EPUB"
     `convert "#{path_to_i}" -colorspace RGB -density #{myres} -resize "200x200>" -quality 100 "#{path_to_i}"`
   end
+  return 'ok'
+rescue => e
+  return e
+end
+
+def convertSpacebreakImgs(imgarr, oebps_dir, logkey='')
+  if imgarr.any?
+    imgarr.each do |i|
+      errcheck = convertSpacebreakImg(oebps_dir, i)
+      # this errcheck is to trap & log (to json) error(s) from the child method
+      if errcheck != "ok"
+        logstring = "error: convertSpacebreakImg() with #{i}: #{errcheck}\n#{logstring}"
+      end
+    end
+  else
+    logstring = 'n-a'
+  end
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def copyFile(src, dest, logkey='')
+	Mcmlln::Tools.copyFile(src, dest)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.runpython in a new method for this script; to return a result for json_logfile
+def localRunPython(py_script, args, logkey='')
+	Bkmkr::Tools.runpython(py_script, args)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def renameFinalEpub(csfilename, stage_dir, logkey='')
+  if stage_dir.include? "egalley" or stage_dir.include? "galley" or stage_dir.include? "firstpass"
+    Mcmlln::Tools.moveFile("#{Bkmkr::Paths.done_dir}/#{Metadata.pisbn}/#{Metadata.eisbn}_EPUB.epub", "#{Bkmkr::Paths.done_dir}/#{Metadata.pisbn}/#{Metadata.eisbn}_EPUBfirstpass.epub")
+    csfilename = "#{Metadata.eisbn}_EPUBfirstpass"
+  end
+  return csfilename
+rescue => logstring
+  return csfilename
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping Bkmkr::Tools.runJar in a new method for this script; to return a result for json_logfile
+def localRunJar(jar_script, input_file, logkey='')
+	Bkmkr::Tools.runjar(jar_script, input_file)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def writeErrfile(epubcheck_output, epubcheck_errfile, logkey='')
+  if epubcheck_output =~ /ERROR/ || epubcheck_output =~ /Check finished with errors/
+  	File.open(epubcheck_errfile, 'w') do |output|
+  		output.puts "Epub validation via epubcheck encountered errors."
+  		output.puts "\n \n(Epubcheck detailed output:)\n "
+  		output.puts epubcheck_output
+  	end
+  else
+    logstring = 'n-a'
+  end
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def sendAlertMail(epubcheck_output, testing_value_file, message, logkey='')
+  if epubcheck_output =~ /ERROR/ || epubcheck_output =~ /Check finished with errors/
+    unless File.file?(testing_value_file)
+      Net::SMTP.start('10.249.0.12') do |smtp|
```

Comment:
We should probably move that IP address to authkeys, and source it in as a variable. It's not really dangerous to have there on it's on, but it does make us a little more vulnerable.